overlord/snapstate/handlers.go

```diff
@@ -1038,9 +1038,12 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	oldCurrent := snapst.Current
 	snapst.Current = cand.Revision
 	snapst.Active = true
-	oldChannel := snapst.Channel
+	oldChannel := snapst.TrackingChannel
 	if snapsup.Channel != "" {
-		snapst.Channel = snapsup.Channel
+		err := snapst.SetTrackingChannel(snapsup.Channel)
```

Comment:
this was discussed but to repeat, because now this can fail we need in follow ups:
* checks in daemon that channel are well-formed
* some cmd/snap code to deal with /stable //foo//bar etc